### PR TITLE
Added file-loader, Clean webpack plugin and Welcome Screen

### PR DESCRIPTION
**Pull request by shubhamsWEB**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 ![Banner](https://user-images.githubusercontent.com/24511864/87848141-f8d04780-c8fa-11ea-8dae-3dde9fe6a68d.png)
 
-# react-enterprise-starter-kit :fire:
+# React-Enterprise-Dtarter-Kit :fire:
```
Comment (anandgupta193) [inline README.md]: please correct typo

File: src/components/App.jsx [hunk 2]
```diff
@@ -1,31 +1,11 @@
...
-      </Button>
-    </div>
-  );
-};
+import React, { lazy, Suspense } from 'react';
+import Loader from './loader/Loader';
+
+const Home = lazy(() => import('./home/Home'));
+const App = () => (
+  <Suspense fallback={<Loader />}>
```
Comment (anandgupta193) [inline src/components/App.jsx]: Suspense should be in routes correct?

File: webpack/webpack.common.config.js [hunk 3]
```diff
@@ -75,6 +76,14 @@ module.exports = {
           },
         ],
       },
+      {
+        test: /\.(png|jpe?g|gif|webp)$/i,
```
Comment (anandgupta193) [inline webpack/webpack.common.config.js]: do we need to add svg's as well here?

File: src/components/home/Home.jsx [hunk 4]
```diff
@@ -1,15 +1,49 @@
...
-      {WELCOME_HEADING}
-    </h1>
-    <Button value={<Link to="/">{BUTTON_TEXT}</Link>} />
+    <a
+      href="https://github.com/anandgupta193/react-enterprise-starter-kit"
+      rel="noreferrer"
+      target="_blank"
+      className="github-corner"
+    >
+      <svg
```
Comment (anandgupta193) [inline src/components/home/Home.jsx]: Let's create a separate folder inside public as svgs and put all the svgs on that, and we will go ahead and import svgs.

**Review by anandgupta193 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (shubhamsWEB) [reply README.md]: Done

File: src/components/App.jsx (hunk 2 above)
Comment (shubhamsWEB) [reply src/components/App.jsx]: right,fixed that

File: webpack/webpack.common.config.js (hunk 3 above)
Comment (shubhamsWEB) [reply webpack/webpack.common.config.js]: Done

File: src/components/home/Home.jsx (hunk 4 above)
Comment (shubhamsWEB) [reply src/components/home/Home.jsx]: For now i have used .png image for the same, but in future we will do that

File: src/components/home/Home.jsx [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
+import ReactLogo from '../../assets/images/React.webp';
 
 const Home = () => (
   <>
-    <h1 className={homeStyle.size60}>
-      {WELCOME_HEADING}
-    </h1>
-    <Button value={<Link to="/">{BUTTON_TEXT}</Link>} />
+    <span className={homeStyle.forkongithub}>
+      <a href="https://github.com/anandgupta193/react-enterprise-starter-kit">
```
Comment (anandgupta193) [inline src/components/home/Home.jsx:9]: get this from constants

File: src/components/home/Home.jsx [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
+import { WELCOME_HEADING } from '../../constants/HomeConstants';
+import ReactLogo from '../../assets/images/React.webp';
 
 const Home = () => (
   <>
-    <h1 className={homeStyle.size60}>
-      {WELCOME_HEADING}
-    </h1>
-    <Button value={<Link to="/">{BUTTON_TEXT}</Link>} />
+    <span className={homeStyle.forkongithub}>
```
Comment (anandgupta193) [inline src/components/home/Home.jsx]: class-names should be camel case - Please fix

File: src/routes/Routes.jsx [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
+const App = lazy(() => import('../components/App'));
 
 const Routes = () => (
   <>
     <BrowserRouter>
       <Switch>
-        <Route path="/" exact component={App} />
-        <Route path="/Home" exact component={Home} />
-        <Route component={notfound} />
+        <Suspense fallback={<Loader />}>
```
Comment (anandgupta193) [inline src/routes/Routes.jsx]: can there be only one suspense for all the routes?

File: webpack/webpack.common.config.js [hunk 8]
```diff
@@ -38,7 +40,8 @@ module.exports = {
           {
             loader: ExtractCssChunks.loader,
             options: {
-              reloadAll: true,
+              hmr: process.env.NODE_ENV === 'production',
```
Comment (anandgupta193) [inline webpack/webpack.common.config.js]: hmr is not handled by webpack-dev-server?

File: webpack/webpack.common.config.js [hunk 9]
```diff
@@ -99,6 +110,6 @@ module.exports = {
   output: {
     filename: env === 'development' ? '[name].js' : '[name].[contenthash].js',
     path: path.resolve(__dirname, '../dist'),
-
+    chunkFilename: 'assets/scripts/[name].[contenthash].js',
```
Comment (anandgupta193) [inline webpack/webpack.common.config.js]: this is amazing

File: src/routes/Routes.jsx [hunk 10]
```diff
@@ -1,16 +1,20 @@
-import React from 'react';
+import React, { lazy, Suspense } from 'react';
 import { BrowserRouter, Route, Switch } from 'react-router-dom';
-import App from '../components/App';
-import Home from '../components/home/Home';
-import notfound from '../components/404/NotFound';
+import Loader from '../components/loader/Loader';
+
+const Notfound = lazy(() => import('../components/404/NotFound'));
```
Comment (anandgupta193) [inline src/routes/Routes.jsx]: As discussed, Can we have separate files for this?

File: src/components/home/Home.jsx [hunk 11]
```diff
@@ -1,15 +1,17 @@
...
+      <a href="https://github.com/anandgupta193/react-enterprise-starter-kit">
+        Fork on GitHub
+      </a>
+    </span>
+    <img className={homeStyle.logo} alt="React-Enterprise-Strater-Kit" src={ReactLogo} />
```
Comment (anandgupta193) [inline src/components/home/Home.jsx]: Get Alt text from constants

**Review by anandgupta193 — CHANGES_REQUESTED** [review]

File: src/components/home/Home.jsx [hunk 12]
```diff
@@ -1,15 +1,18 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
-import Button from '../../pureComponents/Button/Button';
-import { WELCOME_HEADING, BUTTON_TEXT } from '../../constants/HomeConstants';
-import homeStyle from './Home.scss';
+import HomeStyle from './Home.scss';
+import { WELCOME_HEADING, IMG_ALT } from '../../constants/HomeConstants';
+// eslint-disable-next-line import/no-unresolved
```
Comment (anandgupta193) [inline src/components/home/Home.jsx]: why this eslint rule override? cant we fix this?

File: src/routes/Routes.jsx [hunk 13]
```diff
@@ -1,17 +1,17 @@
...
 
 const Routes = () => (
   <>
     <BrowserRouter>
-      <Switch>
-        <Route path="/" exact component={App} />
-        <Route path="/Home" exact component={Home} />
-        <Route component={notfound} />
-      </Switch>
+      <Suspense fallback={<Loader />}>
```
Comment (anandgupta193) [inline src/routes/Routes.jsx:9]: looks amazing

**Comment by anandgupta193** [comment]
@shubhamsWEB - Please rebase

File: src/routes/Routes.jsx [hunk 14]
```diff
@@ -1,17 +1,17 @@
-import React from 'react';
+import React, { Suspense } from 'react';
 import { BrowserRouter, Route, Switch } from 'react-router-dom';
-import App from '../components/App';
-import Home from '../components/home/Home';
-import notfound from '../components/404/NotFound';
+import Loader from '../components/loader/Loader';
+import { Notfound, App } from './Routes.lazy';
```
Comment (anandgupta193) [inline src/routes/Routes.jsx]: Can we change it to NotFound 'F capital'?

**Review by anandgupta193 — APPROVED** [review]

File: src/constants/HomeConstants.js [hunk 15]
```diff
@@ -1,2 +1,2 @@
-export const WELCOME_HEADING = 'Welcome to React Starter Kit';
-export const BUTTON_TEXT = 'BACK';
+export const WELCOME_HEADING = 'React Enterprise Starter Kit';
+export const IMG_ALT = 'React-Enterprise-Strater-Kit';
```
Comment (akhileshbhatia) [inline src/constants/HomeConstants.js]:
```suggestion
export const IMG_ALT = 'React-Enterprise-Starter-Kit';
```

File: .stylelintrc [hunk 16]
```diff
@@ -0,0 +1,6 @@
+{
+  "rules": {
+    'string-quotes': 'single',
+    'no-duplicate-at-import-rules' : true,
+  },
+}
```
Comment (akhileshbhatia) [inline .stylelintrc]: Add a new line

File: .stylelintignore [hunk 17]
```diff
@@ -0,0 +1,5 @@
+node_modules/*
+dist/*
+src/assets/*
+src/**/*.jsx
+src/**/*.js
```
Comment (akhileshbhatia) [inline .stylelintignore]: Add a new line

**Review by akhileshbhatia — CHANGES_REQUESTED** [review]

File: .stylelintignore (hunk 17 above)
Comment (anandgupta193) [reply .stylelintignore]: done

File: .stylelintrc (hunk 16 above)
Comment (anandgupta193) [reply .stylelintrc]: done

File: src/constants/HomeConstants.js (hunk 15 above)
Comment (anandgupta193) [reply src/constants/HomeConstants.js]: done

**Review by akhileshbhatia — APPROVED** [review]
LGTM

**Review by anandgupta193 — APPROVED** [review]

**Pull request merged**